interface.js

```diff
@@ -0,0 +1,436 @@
+console.log('interface.js loaded');
+const Store = require('electron-store');
+const store = new Store();
+
+/* ========================================================================== */
+/* Provider-Specific Handlers                                                 */
+/* ========================================================================== */
+
+/* Shared Handlers ---------------------------------------------------------- */
+
+class Provider {
+	static setupCustomPasteBehavior() {
+		this.webview.addEventListener(
+			'dom-ready',
+			function () {
+				this.webview.executeJavaScript(`
+			document.addEventListener('paste', (event) => {
+				event.preventDefault();
+				let text = event.clipboardData.getData('text');
+				let activeElement = document.activeElement;
+				let start = activeElement.selectionStart;
+				let end = activeElement.selectionEnd;
+				activeElement.value = activeElement.value.slice(0, start) + text + activeElement.value.slice(end);
+				activeElement.selectionStart = activeElement.selectionEnd = start + text.length;
+				});
+			`);
+			}.bind(this)
+		);
+	}
+
+	static handleInput(input) {
+		throw new Error(`Provider ${this.name} must implement handleInput()`);
+	}
+
+	static handleSubmit(input) {
+		throw new Error(`Provider ${this.name} must implement handleSubmit()`);
+	}
+
+	static handleCss() {
+		throw new Error(`Provider ${this.name} must implement handleCss()`);
+	}
+
+	static isEnabled() {
+		throw new Error(`Provider ${this.name} must implement isEnabled()`);
+	}
+}
+
+/* OpenAI ChatGPT ----------------------------------------------------------- */
+
+class OpenAi extends Provider {
+	static webviewId = 'webviewOAI';
+	static webview = document.getElementById('webviewOAI');
+
```

Comment:
prob wanna give it a "display name" too since we'll probably be showing a dropdown